pages/studentAssessments/studentAssessments.sql

```diff
@@ -120,12 +120,17 @@ WITH
       FULL JOIN assessments AS a ON (gc.assessment_id = a.id)
       JOIN course_instances AS ci ON (ci.id = a.course_instance_id)
       JOIN assessment_sets AS aset ON (aset.id = a.assessment_set_id)
-      LEFT JOIN assessment_instances AS ai ON (
-        ai.assessment_id = a.id
-        AND (
-          ai.user_id = $user_id
-          OR ai.group_id = gu.group_id
-        )
+      -- We join assessment_instances twice, once on user_id and once on
+      -- group_id. We used to have these combined with AND (ai.user_id =
+      -- $user_id OR ai.group_id = gu.group_id) but this was triggering a bad
+      -- query plan for some course instances.
```

Comment:
Maybe add slightly more detail about the fact that we have two indexes we want to utilize?

```suggestion
      -- We join assessment_instances twice, once on user_id and once on
      -- group_id. We used to have these combined with AND (ai.user_id =
      -- $user_id OR ai.group_id = gu.group_id) but this was triggering a bad
      -- query plan for some course instances. By joining twice, we encourage
      -- the query planner to utilize the two separate indexes we have.
```